src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs

```diff
@@ -136,7 +136,12 @@ private static PrimarySignature CreatePrimarySignature(CmsSigner cmsSigner, Sign
 
             try
             {
+#if IS_DESKTOP
                 cms.ComputeSignature(cmsSigner);
+#else
+                // In .NET Framework, this parameter is not used and a PIN prompt is always shown. In .NET Core, the silent flag needs to be set to false to show a PIN prompt.
+                cms.ComputeSignature(cmsSigner, silent: false);
```

Comment:
As per [doc](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.pkcs.signedcms.computesignature?view=net-5.0), the difference I see between these overloaded methods is `cms.ComputeSignature(cmsSigner, silent: false);` call can throw `PlatformNotSupportedException`. Do we need to catch this exception along with `CryptographicException`?

Reply:
The doc says that `PlatformNotSupportedException` is thrown when "A signing certificate is not specified" on .NET Core, but on .NET Framework the same condition throws `InvalidOperationException`. We're not catching `InvalidOperationException` here, so the behaviour is consistent between .NET Core and .NET Framework.